### PR TITLE
Tutorials

### PR DESCRIPTION
**Pull request by fabienalet**
This minor PR adds a Python notebook tutorial for the Heisenberg 1d model by showcaing various networks available in NetKet. It also makes very minor modifications to other existing notebooks. What is missing is a tutorial on ConvNets, as well as on MPS.

### CODE REVIEW

File: Tutorials/CustomHamiltonian.ipynb [hunk 1]
```diff
@@ -104,15 +104,15 @@
     "Now, we need to create the relevant bond operators. Since we have two type of terms in our Hamiltonian, we need two different bond operators. To do we simply express the iteraction term as a matrix in the computational basis of the relevant local Hilbert space. In our case, since we are dealing with spin-half degrees of freedom, the local basis of bond is simply $\\lvert \\uparrow \\uparrow \\rangle$, $\\lvert \\uparrow \\downarrow \\rangle$, $\\lvert \\downarrow \\uparrow \\rangle$, $\\lvert \\downarrow \\downarrow \\rangle$. Since\n",
```
Comment (gcarleo) [inline Tutorials/CustomHamiltonian.ipynb]: Some small typos left: 
to do -> to do this (I guess?)
iteraction -> interaction

File: Tutorials/CustomHamiltonian.ipynb [hunk 2]
```diff
@@ -208,12 +208,12 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "Next, we define the Hilbert space but without the constraint of on total magnetization ```total_sz```."
+    "Next, we define the Hilbert space as earlier, except that this time we do not impose the constraint on total magnetization ```total_sz```."
```
Comment (gcarleo) [inline Tutorials/CustomHamiltonian.ipynb]: Maybe we can briefly specify why we don't fix the magnetization? (maybe it's trivial though... )

File: Tutorials/j1j2.ipynb [hunk 3]
```diff
@@ -118,7 +109,7 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "Notice the minus sign in front of the exchange. This is simply a basis rotation corresponndign to the Marshall sign rule to help alleviate the sign problem. The model is still not completely sign problem free due to the next nearest neighbour coupling."
+    "**Side Remark**: Notice the minus sign in front of the exchange. This is simply a basis rotation corresponding to the Marshall sign rule (as an exercise, change the sign of this exchange and observe that the exact diagonalization results in Section 6 do not change). The goal of this basis rotation is to speed up the convergence of the Monte Carlo simulations of the wave-function (by providing a good variational sign structure to start with), but in principle the converged results should be identical in both bases. Note further that this sign change is useful at low frustration (such as here $J_2=0.2$), but may actually be not optimal at stronger frustration. As a bonus exercise, repeat the calculation with $J_2=0.8$, and see which basis (*i.e.* which sign in front of the exchange) leads to faster convergence."
```
Comment (gcarleo) [inline Tutorials/j1j2.ipynb]: Good points!

File: Tutorials/Heisenberg1d.ipynb [hunk 4]
```diff
@@ -0,0 +1,718 @@
...
+    "# Optimizer\n",
+    "op = nk.optimizer.Sgd(learning_rate=0.05)\n",
+    "# Stochastic reconfiguration\n",
+    "gs = nk.variational.Vmc(\n",
+    "    hamiltonian=ha,\n",
+    "    sampler=sa,\n",
+    "    optimizer=op,\n",
+    "    n_samples=1000,\n",
+    "    diag_shift=0.1,\n",
+    "    use_iterative=True,\n",
```
Comment (gcarleo) [inline Tutorials/Heisenberg1d.ipynb]: Maybe it's even faster without the iterative solver on this small system? [not sure]

Comment (kchoo1118) [reply Tutorials/Heisenberg1d.ipynb]: No I think the iterative solver is still faster unless symmetries are used.

File: Tutorials/CustomHamiltonian.ipynb (hunk 1 above)
Comment (fabienalet) [reply Tutorials/CustomHamiltonian.ipynb]: OK done in the commit

File: Tutorials/CustomHamiltonian.ipynb (hunk 2 above)
Comment (fabienalet) [reply Tutorials/CustomHamiltonian.ipynb]: OK done in the commit

**Comment by fabienalet** [comment]
Thanks for the feebacks, I've made the modifications on the tutorials accordingly.

**Pull request merged**